crates/ruff_linter/resources/test/fixtures/pyupgrade/UP046_0.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, AnyStr, Generic, ParamSpec, TypeVar, TypeVarTuple
+
+from somewhere import SupportsRichComparisonT
+
+S = TypeVar("S", str, bytes)  # constrained type variable
+T = TypeVar("T", bound=float)
+Ts = TypeVarTuple("Ts")
+P = ParamSpec("P")
+
+
+class A(Generic[T]):
+    # Comments in a class body are preserved
+    var: T
+
+
+class B(Generic[*Ts]):
+    var: tuple[*Ts]
+
+
+class C(Generic[P]):
+    var: P
+
+
+class Constrained(Generic[S]):
+    var: S
+
+
+# This case gets a diagnostic but not a fix because we can't look up the bounds
+# or constraints on the generic type from another module
+class ExternalType(Generic[T, SupportsRichComparisonT]):
+    var: T
+    compare: SupportsRichComparisonT
+
+
+# typing.AnyStr is a common external type variable, so treat it specially as a
+# known TypeVar
+class MyStr(Generic[AnyStr]):
+    s: AnyStr
+
+
+class MultipleGenerics(Generic[S, T, Ts, P]):
```

Comment:
for a `TypeVarTuple`, this would either be

```suggestion
class MultipleGenerics(Generic[S, T, *Ts, P]):
```

or, if the code was originally written to support Python <3.11

```suggestion
class MultipleGenerics(Generic[S, T, Unpack[Ts], P]):
```

with `Unpack` imported from `typing` or `typing_extensions`. Proper handling of `Unpack` is yet another thing you could possibly handle in a followup PR 😅

Reply:
Ah, I thought I caught all of these this time. Thanks!